pkgs/applications/networking/instant-messengers/teamspeak/client.nix

```diff
@@ -90,34 +62,65 @@ stdenv.mkDerivation rec {
     which
     perl # Installer script needs `shasum`
     qt5.wrapQtAppsHook
+    autoPatchelfHook
   ];
 
+  buildInputs = [
+    zlib
+    glib
+    libpng
+    freetype
+    xorg.libSM
+    xorg.libICE
+    xorg.libXrender
+    openssl
+    xorg.libXrandr
+    xorg.libXfixes
+    xorg.libXcursor
+    xorg.libXinerama
+    xorg.libxcb
+    fontconfig
+    xorg.libXext
+    xorg.libX11
+    alsa-lib
+    libpulseaudio
+    libsForQt5.quazip
+    libcxx
+    libredirect
+    libsForQt5.quazip
+  ] ++ (with qt5; [
+    qtbase
+    qtwebengine
+    qtwebchannel
+    qtsvg
+    qtwebsockets
+  ]);
+
   # This just runs the installer script. If it gets stuck at something like
   # ++ exec
   # + PAGER_PATH=
   # it's looking for a dependency and didn't find it. Check the script and make sure the dep is in nativeBuildInputs.
-  unpackPhase = ''
-    runHook preUnpack
+  buildPhase = ''
+    runHook preBuild
 
     echo -e '\ny' | PAGER=cat sh -xe $src
-    cd TeamSpeak*
 
-    runHook postUnpack
-  '';
+    # Delete unecessary libraries - these are provided by nixos.
```

Comment:
```suggestion
```

Reply:
This is precisely where it's intended to be.